### PR TITLE
feat(interpreter): impl Clone for Stack

### PR DESCRIPTION
**Pull request by jsvisa**
used in https://github.com/paradigmxyz/revm-inspectors/pull/220

### CODE REVIEW

File: crates/interpreter/src/interpreter/stack.rs [hunk 1]
```diff
@@ -7,7 +7,7 @@ use std::vec::Vec;
 pub const STACK_LIMIT: usize = 1024;
 
 /// EVM stack with [STACK_LIMIT] capacity of words.
-#[derive(Debug, PartialEq, Eq, Hash)]
+#[derive(Clone, Debug, PartialEq, Eq, Hash)]
```
Comment (mattsse) [inline crates/interpreter/src/interpreter/stack.rs:10]: this is problematic, because breaks the invariant that this has type hash STACK_LIMIT capacity

Comment (DaniPopes) [reply crates/interpreter/src/interpreter/stack.rs:10]: yeah shouldn't derive but Self::new with copy, pls also add a comment

Comment (jsvisa) [reply crates/interpreter/src/interpreter/stack.rs:10]: How about we implement the clone manually, then we can use `clone()` as usual:

```rust
impl Clone for Stack {
    fn clone(&self) -> Self {
        let mut new_data = Vec::with_capacity(STACK_LIMIT);
        new_data.extend_from_slice(&self.data);
        Self { data: new_data }
    }
}
```

Comment (DaniPopes) [reply crates/interpreter/src/interpreter/stack.rs:10]: that's what I mean

Comment (jsvisa) [reply crates/interpreter/src/interpreter/stack.rs]: @DaniPopes applied, ptal

**Review by DaniPopes — APPROVED** [review]

**Review by mattsse — APPROVED** [review]
lgtm, pending @rakita

**Review by rakita — APPROVED** [review]
lgtm

**Comment by jsvisa** [comment]
@rakita kindly ask when will this feature been released, maybe 19.0.0

**Comment by rakita** [comment]
I hope in two weeks, but next release is going to be a bigger one.

**Comment by rizkyy702** [comment]
> used in https://github.com/paradigmxyz/revm-inspectors/pull/220

**Pull request merged**